diff --git a/ccopts b/ccopts
new file mode 100644
index 0000000..e78267d
--- /dev/null
+++ b/ccopts
@@ -0,0 +1 @@
+-Wall -Wextra -Werror 
diff --git a/main.c b/main.c
new file mode 100644
index 0000000..cc2a095
--- /dev/null
+++ b/main.c
@@ -0,0 +1,30 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   main.c                                             :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: lalgarra <lalgarra@student.42madrid.com>   +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2025/07/09 16:44:38 by lalgarra          #+#    #+#             */
+/*   Updated: 2025/07/12 18:50:32 by lalgarra         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+#include <unistd.h>
+
+void	ft_print_comb2(void);
+void	ft_putnbr(int nb);
+
+int	main(void)
+{
+	ft_print_comb2();
+	write(1, "\n\n= = = ex07: 42\n", 17);
+	ft_putnbr(42);
+	write(1, "\n\n= = = ex07: 0\n", 16);
+	ft_putnbr(0);
+	write(1, "\n\n= = = ex07: 32321\n", 20);
+	ft_putnbr(32321);
+	write(1, "\n\n= = = ex07: -20\n", 18);
+	ft_putnbr(-20);
+	write(1, "\n\n= = = ex08\n", 13);
+	return (0);
+}
